Tests/Simulation/SimulatedLoadTests.cs

```diff
@@ -39,10 +42,11 @@ public SimulatedLoadTests(IntegrationTestFixtureSim testFixture, ITestOutputHelp
             this.uniqueMessageFragment = Guid.NewGuid().ToString();
             this.logger = new TestOutputLogger(testOutputHelper);
             this.simulatedBasicsStations =
-                testFixture.DeviceRange5000_BasicsStationSimulators
-                           .Select((basicsStation, i) => (BasicsStation: basicsStation, Index: i))
-                           .Select(b => new SimulatedBasicsStation(StationEui.Parse(b.BasicsStation.DeviceID), Configuration.LnsEndpointsForSimulator[b.Index % Configuration.LnsEndpointsForSimulator.Count]))
+                testFixture.DeviceRange5000_BasicsStationSimulators.Index()
+                           .Select(b => new SimulatedBasicsStation(StationEui.Parse(b.Value.DeviceID), Configuration.LnsEndpointsForSimulator[b.Key % Configuration.LnsEndpointsForSimulator.Count]))
```

Comment:
nit: could use `Zip` to pair devices with LNS URLs that cycle via [`Repeat`](https://morelinq.github.io/3.3/ref/api/html/M_MoreLinq_MoreEnumerable_Repeat__1.htm):

```diff
diff --git a/Tests/Simulation/SimulatedLoadTests.cs b/Tests/Simulation/SimulatedLoadTests.cs
index 99941234..e0f713d1 100644
--- a/Tests/Simulation/SimulatedLoadTests.cs
+++ b/Tests/Simulation/SimulatedLoadTests.cs
@@ -19,6 +19,7 @@ namespace LoRaWan.Tests.Simulation
     using Xunit;
     using Xunit.Abstractions;
     using static MoreLinq.Extensions.IndexExtension;
+    using static MoreLinq.Extensions.RepeatExtension;
     using static MoreLinq.Extensions.TransposeExtension;
 
     [Trait("Category", "SkipWhenLiveUnitTesting")]
@@ -42,8 +43,9 @@ namespace LoRaWan.Tests.Simulation
             this.uniqueMessageFragment = Guid.NewGuid().ToString();
             this.logger = new TestOutputLogger(testOutputHelper);
             this.simulatedBasicsStations =
-                testFixture.DeviceRange5000_BasicsStationSimulators.Index()
-                           .Select(b => new SimulatedBasicsStation(StationEui.Parse(b.Value.DeviceID), Configuration.LnsEndpointsForSimulator[b.Key % Configuration.LnsEndpointsForSimulator.Count]))
+                testFixture.DeviceRange5000_BasicsStationSimulators
+                           .Zip(Configuration.LnsEndpointsForSimulator.Repeat(),
+                                (dev, lnsUrl) => new SimulatedBasicsStation(StationEui.Parse(dev.DeviceID), lnsUrl))
                            .ToList();
 
             Assert.True(this.simulatedBasicsStations.Count % Configuration.LnsEndpointsForSimulator.Count == 0, "Since Basics Stations are round-robin distributed to LNS, we must have the same number of stations per LNS for well-defined test assertions.");
```

Reply:
Done with d19715d67624762fb2bfe3258c4d93193cc0d6df.